db/user.go

```diff
@@ -221,3 +222,71 @@ func SyncUsers(course *model.Course, newUsers map[string]*model.User,
 
     return syncResult, nil;
 }
+
+// ResolveUsers is a helper function to maps string representations of roles and * (all roles) to the emails for users with those roles.
+// The function takes an optional course and a list of strings, containing emails, roles, and * as input and returns an alphanumerically sorted slice of lowercase emails without duplicates.
+// A fatal error occurs if a course cannot be opened and the input contains roles or a *.
+// A warning will be logged for every invalid role.
+func ResolveUsers(course *model.Course, emails []string) ([]string, error) {
+    if (backend == nil) {
+        return nil, fmt.Errorf("Database has not been opened.");
+    }
+
+    emailSet := map[string]any{};
+    roleSet := map[string]any{};
+
+    // Iterate over every given string, checking for emails, roles, and *, which denotes all users.
```

Comment:
Use parens for the all users aside (a comma here, although correct, is confusing since it appears at the end of a sentence with a list.

s/, which denotes all users./ (which denotes all users)./